Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -97,9 +107,9 @@ public int DownloadedBlocksCount
             }
         }
 
-        /// <summary>Sets of block header hashes that are being downloaded mapped by peers they are assigned to.</summary>
+        /// <summary>Sets of download tasks representing blocks that are being downloaded mapped by peers they are assigned to.</summary>
         /// <remarks>All access to this object has to be protected by <see cref="lockObject"/>.</remarks>
-        private readonly Dictionary<BlockPullerBehavior, HashSet<uint256>> peersPendingDownloads = new Dictionary<BlockPullerBehavior, HashSet<uint256>>();
+        private readonly Dictionary<BlockPullerBehavior, Dictionary<uint256, DownloadTask>> peersPendingDownloads = new Dictionary<BlockPullerBehavior, Dictionary<uint256, DownloadTask>>();
```

Comment:
Hmm that's a pity, I think it should not affect our ability to use this word. But if you are strong on this, I'll change it to DownloadRequest or DownloadAssignment. It is not DownloadMessage though.